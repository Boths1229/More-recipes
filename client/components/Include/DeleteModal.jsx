import React from 'react';

const DeleteModal = (props) => {
  const { deleteRecipe, recipeId } = props;
  console.log("Iiiiiiiii", props);
  return (
    <div
      className="modal fade"
      id="exampleModal10"
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel1">
              Delete recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            This action is irrevesible, pls process with caution
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              onClick={() => deleteRecipe(recipeId)}
              className="btn btn-danger"
              data-dismiss="modal"
            >
              Delete recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DeleteModal;
