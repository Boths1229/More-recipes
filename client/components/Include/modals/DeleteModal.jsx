import React from 'react';
import PropTypes from 'prop-types';

/**
 * @description functional component to render the delete recipe confirmation modal
 *
 * @param {object} props
 *
 * @returns {JSX} JSX
 */
const DeleteModal = (props) => {
  const { deleteRecipe, recipeId } = props;
  return (
    <div
      className="modal fade"
      id={`deleteModal-${recipeId}`}
      tabIndex="-1"
      role="dialog"
      aria-labelledby="exampleModalLabel"
      aria-hidden="true"
    >
      <div className="modal-dialog" role="document">
        <div className="modal-content">
          <div className="modal-header">
            <h5 className="modal-title" id="exampleModalLabel1">
              Delete recipe
            </h5>
            <button
              type="button"
              className="close"
              data-dismiss="modal"
              aria-label="Close"
            >
              <span aria-hidden="true">&times;</span>
            </button>
          </div>
          <div className="modal-body">
            This action is irrevesible, pls process with caution
          </div>
          <div className="modal-footer">
            <button
              type="button"
              className="btn btn-secondary"
              data-dismiss="modal"
            >
              Close
            </button>
            <button
              type="button"
              onClick={() => deleteRecipe(recipeId)}
              className="btn btn-danger"
              data-dismiss="modal"
            >
              Delete recipe
            </button>
          </div>
        </div>
      </div>
    </div>
  );
};

DeleteModal.propTypes = {
  deleteRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired
};

export default DeleteModal;
