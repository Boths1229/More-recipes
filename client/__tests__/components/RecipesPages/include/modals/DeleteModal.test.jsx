import React from 'react';
import { shallow } from 'enzyme';
import DeleteModal from '../../../../../components/Include/modals/DeleteModal';

describe('Delete modal component', () => {
  it('matches snapshot', () => {
    const props = {
      value: {}
    };
    const output = shallow(<DeleteModal {...props} />);
    expect(output).toMatchSnapshot();
  });
  it('calls onClick', () => {
    const props = {
      value: {},
      deleteRecipe: jest.fn()
    };
    const output = shallow(<DeleteModal {...props} />);
    const button = output.find('button[className="btn btn-danger"]');
    button.simulate('click');
    expect(props.deleteRecipe).toHaveBeenCalled();
  });
});
