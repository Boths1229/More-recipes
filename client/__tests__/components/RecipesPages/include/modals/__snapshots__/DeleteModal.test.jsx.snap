// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete modal component matches snapshot 1`] = `
<div
  aria-hidden="true"
  aria-labelledby="exampleModalLabel"
  className="modal fade"
  id="deleteModal-undefined"
  role="dialog"
  tabIndex="-1"
>
  <div
    className="modal-dialog"
    role="document"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <h5
          className="modal-title"
          id="exampleModalLabel1"
        >
          Delete recipe
        </h5>
        <button
          aria-label="Close"
          className="close"
          data-dismiss="modal"
          type="button"
        >
          <span
            aria-hidden="true"
          >
            Ã—
          </span>
        </button>
      </div>
      <div
        className="modal-body"
      >
        This action is irrevesible, pls process with caution
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-secondary"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
        <button
          className="btn btn-danger"
          data-dismiss="modal"
          id="del"
          onClick={[Function]}
          type="button"
        >
          Delete recipe
        </button>
      </div>
    </div>
  </div>
</div>
`;
